--- ray-ray-0.2.1/python/ray/services.py	2017-11-27 02:15:06.006324415 +0000
+++ ray-ray-0.2.1-new/python/ray/services.py	2017-11-27 02:14:42.265284166 +0000
@@ -1108,7 +1108,7 @@
                    num_gpus=None,
                    num_custom_resource=None,
                    num_redis_shards=None,
-                   include_webui=True,
+                   include_webui=False,
                    plasma_directory=None,
                    huge_pages=False):
     """Start Ray in local mode.
