--- ray-ray-0.2.1/python/ray/scripts/scripts.py	2017-11-27 02:15:06.006324415 +0000
+++ ray-ray-0.2.1-new/python/ray/scripts/scripts.py	2017-11-27 02:14:42.265284166 +0000
@@ -178,29 +178,30 @@
 
 @click.command()
 def stop():
-    subprocess.call(["killall global_scheduler plasma_store plasma_manager "
-                     "local_scheduler"], shell=True)
+    subprocess.call(["killall local_scheduler"], shell=True)
+    subprocess.call(["killall plasma_manager"], shell=True)
+    subprocess.call(["killall global_scheduler plasma_store"], shell=True)
 
     # Find the PID of the monitor process and kill it.
-    subprocess.call(["kill $(ps aux | grep monitor.py | grep -v grep | "
-                     "awk '{ print $2 }') 2> /dev/null"], shell=True)
+    subprocess.call(["kill -s 15 $(ps | grep monitor.py | grep -v grep | "
+                     "awk '{ print $1 }') 2> /dev/null"], shell=True)
 
     # Find the PID of the Redis process and kill it.
-    subprocess.call(["kill $(ps aux | grep redis-server | grep -v grep | "
-                     "awk '{ print $2 }') 2> /dev/null"], shell=True)
+    subprocess.call(["kill -s 15 $(ps | grep redis-server | grep -v grep | "
+                     "awk '{ print $1 }') 2> /dev/null"], shell=True)
 
     # Find the PIDs of the worker processes and kill them.
-    subprocess.call(["kill -9 $(ps aux | grep default_worker.py | "
-                     "grep -v grep | awk '{ print $2 }') 2> /dev/null"],
+    subprocess.call(["kill -9 $(ps | grep default_worker.py | "
+                     "grep -v grep | awk '{ print $1 }') 2> /dev/null"],
                     shell=True)
 
     # Find the PID of the Ray log monitor process and kill it.
-    subprocess.call(["kill $(ps aux | grep log_monitor.py | grep -v grep | "
-                     "awk '{ print $2 }') 2> /dev/null"], shell=True)
+    subprocess.call(["kill -s 15 $(ps | grep log_monitor.py | grep -v grep | "
+                     "awk '{ print $1 }') 2> /dev/null"], shell=True)
 
     # Find the PID of the jupyter process and kill it.
-    subprocess.call(["kill $(ps aux | grep jupyter | grep -v grep | "
-                     "awk '{ print $2 }') 2> /dev/null"], shell=True)
+    subprocess.call(["kill $(ps | grep jupyter | grep -v grep | "
+                     "awk '{ print $1 }') 2> /dev/null"], shell=True)
 
 
 cli.add_command(start)
