--- gnu-config-20120814/config.sub	2012-09-18 10:27:45.000000000 +0100
+++ /home/martin/config.sub	2014-09-23 23:14:10.619326512 +0100
@@ -311,6 +312,7 @@ case $basic_machine in
 	| sparc | sparc64 | sparc64b | sparc64v | sparc86x | sparclet | sparclite \
 	| sparcv8 | sparcv9 | sparcv9b | sparcv9v \
 	| spu \
+	| riscv \
 	| tahoe | tic4x | tic54x | tic55x | tic6x | tic80 | tron \
 	| ubicom32 \
 	| v850 | v850e | v850e1 | v850e2 | v850es | v850e2v3 \
@@ -1027,6 +1029,11 @@ case $basic_machine in
 		basic_machine=i386-pc
 		os=-rdos
 		;;
+	riscv)
+		basic_machine=riscv
+		;;
+	riscv-*)
+		;;
 	rom68k)
 		basic_machine=m68k-rom68k
 		os=-coff
