--- ray-ray-0.2.1/src/common/CMakeLists.txt	2017-10-01 19:33:13.000000000 +0000
+++ ray-ray-0.2.1-new/src/common/CMakeLists.txt	2017-11-27 20:11:36.743140130 +0000
@@ -2,6 +2,8 @@
 
 project(common)
 
+set(CMAKE_INCLUDE_SYSTEM_FLAG_CXX "-I ") #for gcc 6 and higher
+
 include(${CMAKE_CURRENT_LIST_DIR}/cmake/Common.cmake)
 
 add_subdirectory(redis_module)
@@ -48,7 +50,7 @@
 
 add_custom_target(
    hiredis
-   COMMAND make
+   COMMAND make VERBOSE=1 
    WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/thirdparty/hiredis)
 
 add_library(common STATIC
@@ -73,12 +75,14 @@
 
 add_dependencies(common gen_common_fbs)
 
+set(CMAKE_LINKER ${CMAKE_CURRENT_LIST_DIR}/../../../recipe-sysroot-native/usr/bin/riscv64-poky-linux/riscv64-poky-linux-ld)
 target_link_libraries(common "${CMAKE_CURRENT_LIST_DIR}/thirdparty/hiredis/libhiredis.a")
 
 function(define_test test_name library)
   add_executable(${test_name} test/${test_name}.cc ${ARGN})
   add_dependencies(${test_name} hiredis flatbuffers_ep)
-  target_link_libraries(${test_name} common ${FLATBUFFERS_STATIC_LIB} ${ARROW_DIR}/cpp/build/release/libarrow.a ${library} -lpthread)
+  set(ARROW_DIR ${CMAKE_CURRENT_LIST_DIR}/../../../../../python-pyarrow/0.7.1-r0/recipe-sysroot/usr/)
+  target_link_libraries(${test_name} common ${FLATBUFFERS_STATIC_LIB} ${ARROW_DIR}/lib/libarrow.a ${library} -lpthread)
   target_compile_options(${test_name} PUBLIC "-DPLASMA_TEST -DLOCAL_SCHEDULER_TEST -DCOMMON_TEST -DRAY_COMMON_LOG_LEVEL=4")
 endfunction()
 
@@ -90,16 +94,17 @@
 define_test(task_table_tests "")
 define_test(object_table_tests "")
 
-add_custom_target(
-   redis
-   COMMAND make
-   WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/thirdparty/redis)
+#add_custom_target(
+#   redis
+#   COMMAND make
+#   WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/thirdparty/redis)
 
 add_custom_target(copy_redis ALL)
 foreach(file "redis-cli" "redis-server")
 add_custom_command(TARGET copy_redis POST_BUILD
                    COMMAND ${CMAKE_COMMAND} -E
-                      copy ${CMAKE_CURRENT_LIST_DIR}/thirdparty/redis/src/${file}
+                      #copy ${CMAKE_CURRENT_LIST_DIR}/thirdparty/redis/src/${file}
+                      copy ${CMAKE_CURRENT_LIST_DIR}/../../../../../redis/4.0.2-r0/image/usr/bin/${file}
                            ${CMAKE_BINARY_DIR}/src/common/thirdparty/redis/src/${file})
 endforeach()
-add_dependencies(copy_redis redis)
+#add_dependencies(copy_redis redis)
