--- ray-ray-0.2.1/src/common/thirdparty/hiredis/Makefile	2017-10-01 19:33:13.000000000 +0000
+++ ray-ray-0.2.1-new/src/common/thirdparty/hiredis/Makefile	2017-11-27 08:49:13.321197957 +0000
@@ -9,6 +9,15 @@
 LIBNAME=libhiredis
 PKGCONFNAME=hiredis.pc
 
+mkfile_path := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+CC=$(mkfile_path)/../../../../../recipe-sysroot-native/usr/bin/riscv64-poky-linux/riscv64-poky-linux-gcc 
+CXX=$(mkfile_path)/../../../../..//recipe-sysroot-native/usr/bin/riscv64-poky-linux/riscv64-poky-linux-g++ 
+CFLAGS=--sysroot=$(mkfile_path)/../../../../../recipe-sysroot
+LDFLAGS=--sysroot=$(mkfile_path)/../../../../../recipe-sysroot/ -L$(mkfile_path)/../../../../../recipe-sysroot/lib -L$(mkfile_path)/../../../../../recipe-sysroot/usr/lib -L$(mkfile_path)/../../../../../recipe-sysroot/lib -L$(mkfile_path)/../../../../../recipe-sysroot/usr/lib/riscv64-poky-linux/7.2.0/
+
+
+#echo library path is $$LD_LIBRARY_PATH; \
+
 HIREDIS_MAJOR=$(shell grep HIREDIS_MAJOR hiredis.h | awk '{print $$3}')
 HIREDIS_MINOR=$(shell grep HIREDIS_MINOR hiredis.h | awk '{print $$3}')
 HIREDIS_PATCH=$(shell grep HIREDIS_PATCH hiredis.h | awk '{print $$3}')
@@ -51,7 +60,9 @@
 DYLIBNAME=$(LIBNAME).$(DYLIBSUFFIX)
 DYLIB_MAKE_CMD=$(CC) -shared -Wl,-soname,$(DYLIB_MINOR_NAME) -o $(DYLIBNAME) $(LDFLAGS)
 STLIBNAME=$(LIBNAME).$(STLIBSUFFIX)
-STLIB_MAKE_CMD=ar rcs $(STLIBNAME)
+STLIB_CMD=$(mkfile_path)/../../../../../recipe-sysroot-native/usr/bin/riscv64-poky-linux/riscv64-poky-linux-gcc-ar rcs
+#STLIB_CMD=ar rcs
+STLIB_MAKE_CMD=$(STLIB_CMD) $(STLIBNAME)
 
 # Platform-specific overrides
 uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
@@ -78,6 +89,7 @@
 test.o: test.c fmacros.h hiredis.h read.h sds.h
 
 $(DYLIBNAME): $(OBJ)
+	export LD_LIBRARY_PATH=$(mkfile_path)/../../../../../recipe-sysroot/usr/lib/riscv64-poky-linux/7.2.0/:$(mkfile_path)/../../../../../recipe-sysroot/lib:$(mkfile_path)/../../../../../recipe-sysroot/usr/lib:$(mkfile_path)/../../../../../recipe-sysroot/usr/lib/riscv64-poky-linux/7.2.0/; \
 	$(DYLIB_MAKE_CMD) $(OBJ)
 
 $(STLIBNAME): $(OBJ)
