--- arrow-apache-arrow-0.7.1/python/CMakeLists.txt	2017-09-27 13:19:05.000000000 +0000
+++ arrow-apache-arrow-0.7.1-new/python/CMakeLists.txt	2017-11-27 05:14:58.195860480 +0000
@@ -22,9 +22,12 @@
 project(pyarrow)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake_modules")
+set(CMAKE_INCLUDE_SYSTEM_FLAG_CXX "-I ") #for gcc 6 and higher
 
 include(CMakeParseArguments)
 
+set( PKG_CONFIG_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../recipe-sysroot/usr/lib/pkgconfig" )
+
 # Compatibility with CMake 3.1
 if(POLICY CMP0054)
   # http://www.cmake.org/cmake/help/v3.1/policy/CMP0054.html
@@ -315,7 +318,8 @@
     libplasma_shared)
 
   set(CYTHON_EXTENSIONS ${CYTHON_EXTENSIONS} plasma)
-  file(COPY ${PLASMA_EXECUTABLE} DESTINATION ${BUILD_OUTPUT_ROOT_DIRECTORY})
+  #file(COPY ${PLASMA_EXECUTABLE} DESTINATION ${BUILD_OUTPUT_ROOT_DIRECTORY})
+  file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/../../../../apache-arrow/0.7.1-r0/image/usr/bin/plasma-dir/plasma_store DESTINATION ${BUILD_OUTPUT_ROOT_DIRECTORY})
 endif()
 
 ############################################################
